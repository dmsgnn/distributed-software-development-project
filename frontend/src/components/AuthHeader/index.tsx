import { Box, BoxProps, Flex, Heading, Link, Text } from '@chakra-ui/react';
import NextLink from 'next/link';
import { useRouter } from 'next/router';

const AuthHeaderLink = ({ href, children }) => {
	const router = useRouter();

	return (
		<Box
			textAlign="center"
			flex={1}
			borderBottom="2px"
			borderColor="blue.100"
			aria-current={router.pathname === href ? 'page' : undefined}
			_activeLink={{
				borderColor: 'blue.500',
			}}
		>
			<Link as={NextLink} href={href} role={undefined}>
				{children}
			</Link>
		</Box>
	);
};

export default function AuthHeader(props: BoxProps) {
	return (
		<Box textAlign="center" {...props}>
			<Heading as="b">DSEC</Heading>
			<Text>
				The simplest way to create <b>secure software</b>
			</Text>

			<Flex mt={4}>
				<AuthHeaderLink href="/login">Login</AuthHeaderLink>
				<AuthHeaderLink href="/register">Register</AuthHeaderLink>
			</Flex>
		</Box>
	);
}
