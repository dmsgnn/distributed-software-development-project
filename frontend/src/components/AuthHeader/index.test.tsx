import { axe } from 'jest-axe';
import { useRouter } from 'next/router';

import { act, render } from 'test-utils';
import AuthHeader from './';

jest.mock('next/router', () => ({ useRouter: jest.fn(() => ({ pathname: '/' })) }));

describe('AuthHeader', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<AuthHeader />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<AuthHeader />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});

	it('should be highlight current page', async () => {
		(useRouter as jest.Mock).mockReturnValue({ pathname: '/login' });
		const { asFragment } = render(<AuthHeader />);

		expect(asFragment()).toMatchSnapshot();
	});
});
