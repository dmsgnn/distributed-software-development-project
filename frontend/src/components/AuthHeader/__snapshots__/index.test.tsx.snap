// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader should be highlight current page 1`] = `
<DocumentFragment>
  <div
    class="css-xi606m"
  >
    <b
      class="chakra-heading css-0"
    >
      DSEC
    </b>
    <p
      class="chakra-text css-0"
    >
      The simplest way to create 
      <b>
        secure software
      </b>
    </p>
    <div
      class="css-1n0zz00"
    >
      <div
        aria-current="page"
        class="css-1hxhl85"
      >
        <a
          class="chakra-link css-0"
          href="/login"
        >
          Login
        </a>
      </div>
      <div
        class="css-1hxhl85"
      >
        <a
          class="chakra-link css-0"
          href="/register"
        >
          Register
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthHeader should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-xi606m"
  >
    <b
      class="chakra-heading css-0"
    >
      DSEC
    </b>
    <p
      class="chakra-text css-0"
    >
      The simplest way to create 
      <b>
        secure software
      </b>
    </p>
    <div
      class="css-1n0zz00"
    >
      <div
        class="css-1hxhl85"
      >
        <a
          class="chakra-link css-0"
          href="/login"
        >
          Login
        </a>
      </div>
      <div
        class="css-1hxhl85"
      >
        <a
          class="chakra-link css-0"
          href="/register"
        >
          Register
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
